enhancements/windows-containers/csi-proxy.md

```diff
@@ -0,0 +1,160 @@
+storage drivers, [which are in the process of being deprecated and removed.](https://kubernetes.io/blog/2022/09/26/storage-in-tree-to-csi-migration-status-update-1.25/)
+
+This will be done by running [CSI Proxy](https://github.com/kubernetes-csi/csi-proxy) as a Windows service on each
+Windows node. csi-proxy is a pre-requisite for CSI node driver pods to interact with the underlying Windows storage
+APIs. With csi-proxy running on a cluster's Windows nodes, cluster administrators are free to deploy the CSI storage
+drivers of their choice as a Windows DaemonSet, enabling Windows pods on each node to have access to storage through
+PersistentVolumeClaims.
+
+## Motivation
+
+Windows nodes on OpenShift have had persistent storage available through the use of the in-tree storage drivers.
+Persistent storage is a necessary feature for production workloads on Kubernetes, and it is important that this
+functionality continue to work, even with the in-tree drivers being removed.
+
+### User Stories
+
+* As a developer using OpenShift, I want the Windows workloads I create to be able to share storage between workloads,
+  and have that storage persist when pods are removed and re-created.
+
+### Goals
+
+* Windows nodes configured by WMCO fulfill all the pre-requisites to run CSI node drivers.
+
+### Non-Goals
+
+* The Windows CSI storage drivers will not be distributed as part of OpenShift.
+  * At this time, we are not able to build or ship Windows images due to support restrictions.
+
+## Proposal
+
+The [windows-machine-config-operator](https://github.com/openshift/windows-machine-config-operator/)(WMCO),
+will ensure that csi-proxy is running on the Windows nodes it configures. This will be done through the normal
+workflow for adding a service to the expected service list.
+
+The csi-proxy binary will be baked into the WMCO image. WMCO will copy the binary to each node at configuration
+time. A new item will be added to the windows-services ConfigMap, instructing the windows-instance-config-daemon
+to ensure that csi-proxy is always running as a Windows service on each node.
+
+With the nodes in this state, users are free to follow upstream documentation to deploy the CSI node drivers
+of their choice on the Windows nodes in the cluster.
+
+No support from Red Hat will be given for issues occurring with the upstream drivers, as they are not being distributed
+as part of OpenShift.
+
+
+### Workflow Description
+
+1. The cluster administrator will identify what CSI storage drivers the cluster requires for its workloads.
+2. The cluster admin can follow upstream documentation to deploy the CSI node drivers on its Windows nodes.
+
+### API Extensions
+
+N/A
+
+### Risks and Mitigations
+
+If there are any security issues present in the CSI driver images deployed by the user, they will not be fixed
+by upgrading OCP. It is up to the user to stay informed and update their images as necessary.
```

Comment:
- Isn't there a risk of an upgrade potentially breaking a driver?
- What about a mismatch in versions between Linux and Windows? You should link to the version skew section here.